snakesist/exist_client.py

```diff
@@ -101,6 +83,77 @@ def node_id(self):
         return self._node_id
 
 
+class DocumentResource(Resource):
+    """
+    A representation of an eXist document node
+    """
+    def __init__(
+            self,
+            exist_client: "ExistClient",
+            query_result: Optional[QueryResultItem] = None
+    ):
+        """
+        :param exist_client: The client to which the resource is coupled.
+        :query_result: A tuple containing the absolute resource ID, node ID
+                       and the node of the resource.
+        """
+        super().__init__(exist_client, query_result)
+
+    def delete(self):
+        """
+        Remove the document from the database.
+        """
+        self._exist_client.delete_document(path=self.path)
+        self._node_id = None
```

Comment:
these both should be empty strings, shouldn't they? what does `mypy` say?

Reply:
`mypy` points it out, of course. i just forgot to use it.